src/style_context.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013-2020, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::object::IsA;
+use glib::translate::*;
+use gtk_sys;
+use libc::c_void;
+use pango::FontDescription;
+use pango_sys::PangoFontDescription;
+use StyleContext;
+
+pub trait StyleContextExtManual: 'static {
+    fn get_font(&self) -> Option<FontDescription>;
+}
+
+impl<O: IsA<StyleContext>> StyleContextExtManual for O {
+    fn get_font(&self) -> Option<FontDescription> {
+        unsafe {
+            let font_description_ptr: *mut PangoFontDescription = std::ptr::null_mut();
+            gtk_sys::gtk_style_context_get(
+                self.as_ref().to_glib_none().0,
+                gtk_sys::GTK_STATE_FLAG_NORMAL,
```

Comment:
Shouldn't the state flags also be passed into the function by the caller?

Reply:
You right :(

Reply:
The getter for border, color, etc have that at least.

Reply:
Fixed